makefiles/suit.base.inc.mk

```diff
@@ -18,39 +22,47 @@ else
   SUIT_KEY_DIR ?= $(XDG_DATA_HOME)/RIOT/keys
 endif
 
-SUIT_SEC ?= $(SUIT_KEY_DIR)/$(SUIT_KEY).pem
+# we may accept multiple keys for the firmware
+SUIT_SEC ?= $(foreach item,$(SUIT_KEY),$(SUIT_KEY_DIR)/$(item).pem)
+# but there can only be one signing key
+SUIT_SEC_SIGN ?= $(SUIT_KEY_DIR)/$(SUIT_KEY_SIGN).pem
 
-# Multiple keys can be specified with "key0:pw0 key1:pw1 …" (pw may be empty)
-SUIT_SECS ?= $(SUIT_SEC):$(SUIT_SEC_PASSWORD)
+# generate a list of the public keys
+SUIT_PUBS ?= $(SUIT_SEC:.pem=.pem.pub)
 
 SUIT_PUB_HDR = $(BINDIR)/riotbuild/public_key.h
 SUIT_PUB_HDR_DIR = $(dir $(SUIT_PUB_HDR))
 CFLAGS += -I$(SUIT_PUB_HDR_DIR)
 BUILDDEPS += $(SUIT_PUB_HDR)
 
+ifneq (,$(SUIT_SEC_PASSWORD))
```

Comment:
If the user has to enter the password anyway for a new key, he could also select the encryption.
I think this should be independent of whether `SUIT_SEC_PASSWORD` has any value.

```diff
diff --git a/makefiles/suit.base.inc.mk b/makefiles/suit.base.inc.mk
index 58f116c976..e5035e1de6 100644
--- a/makefiles/suit.base.inc.mk
+++ b/makefiles/suit.base.inc.mk
@@ -35,16 +35,29 @@ SUIT_PUB_HDR_DIR = $(dir $(SUIT_PUB_HDR))
 CFLAGS += -I$(SUIT_PUB_HDR_DIR)
 BUILDDEPS += $(SUIT_PUB_HDR)
 
-ifneq (,$(SUIT_SEC_PASSWORD))
-  OPENSSL_GENPKEY_ARGS += -aes-256-cbc
-endif
-
 # OpenSSL leaves an empty file if key generation fails - remove it manually
 # see https://github.com/openssl/openssl/issues/25440
 $(SUIT_SEC): | $(CLEAN)
        $(Q)echo suit: generating key in $(SUIT_KEY_DIR)
        $(Q)mkdir -p $(SUIT_KEY_DIR)
-       -$(Q)openssl genpkey -algorithm ed25519 $(OPENSSL_GENPKEY_ARGS) -out $@
+       $(Q)(   \
+  printf "0) none\n";  \
+  printf "1) aes-256-cbc\n";   \
+       printf "Choose encryption for key file $@: ";   \
+       read encryption;        \
+       case $$encryption in    \
+               0)      \
+                       openssl genpkey -algorithm ed25519 -out $@;     \
+                       ;;      \
+               1)      \
+                       openssl genpkey -algorithm ed25519 -aes-256-cbc -out $@;        \
+                       ;;      \
+               *)      \
+                       echo "Invalid choice";  \
+                       exit 1; \
+                       ;;      \
+       esac;   \
+       )
        $(Q)if [ ! -s $@ ]; then rm $@; fi
 
 %.pem.pub: %.pem
```

This is not using `select` because it s a `bash` only feature.

Reply:
This is much nicer!
The only issue is that this now leaves an empty key file behind if you don't enter the same password twice on key generation.
On the next run, `make` then things there already is a key and fails with an unintuitive error :confused: